spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainerIntegrationTests.java

```diff
@@ -372,6 +382,7 @@ public void testAddRemoveConsumers() throws Exception {
 		assertThat(activeConsumerCount(container, 0)).isTrue();
 		assertThat(TestUtils.getPropertyValue(container, "consumersByQueue", MultiValueMap.class)).hasSize(0);
 		template.stop();
+		assertThat(listenerThreadNames).contains("myThread"); // BUT IT'S STILL "client-1"
```

Comment:
Could you, please, advise me, on how to do it better here? When there is a CachingConnectionFactory with a ThreadPoolTaskExecutor which has a threadNamePrefix, it is used when creating a connection thread and then this thread is used for consuming the message. It would be ok to set threadNamePrefix in the DirectMessageListenerContainer the same way it is done here in tests, but ConnectionFactory interface doesn’t have getters\setters for the executer.
Here, I’ve added some things to the tests which I expect from it, but it doesn’t work.